ISISComputingGroup/EPICS-inst_servers
# Ticket4110 blockserver

### Description of work

Fix a bug in the blockserver where configs had different structures depending on whether they were set as JSON (from the GUI) or read from XML (on the filesystem)

I've also done a fair bit of general tidying of the code, in terms of trying to make it easier to understand.

### To test

https://github.com/ISISComputingGroup/IBEX/issues/4110

I believe I have also fixed the following tickets. Reviewer, please verify and close these if appropriate:
- [x] https://github.com/ISISComputingGroup/IBEX/issues/4061
- [x] https://github.com/ISISComputingGroup/IBEX/issues/4150
- [x] https://github.com/ISISComputingGroup/IBEX/issues/4006

### Acceptance criteria

Bug described in https://github.com/ISISComputingGroup/IBEX/issues/4090 is fixed with a better solution than just returning True.

---

#### Code Review


### Functional Tests


### Final steps
